plotly/tools.py

```diff
@@ -1443,6 +1443,376 @@ class FigureFactory(object):
+                                 title = "Scatterplot Matrix"):
+
+        from plotly import tools
+        import plotly.plotly as py
+        import plotly.graph_objs as go
+        import pandas as pd
+        import random as r
+       
+        colors = ["rgb(31, 119, 180)", "rgb(255, 127, 14)",
+                  "rgb(44, 160, 44)", "rgb(214, 39, 40)",
+                  "rgb(148, 103, 189)", "rgb(140, 86, 75)",
+                  "rgb(227, 119, 194)", "rgb(127, 127, 127)",
+                  "rgb(188, 189, 34)", "rgb(23, 190, 207)"]
+
+        matrix = []
+        col_names = []
+        index_vals = []
+        diagonal_choices = ["Scatter", "Histogram"]
+
+        # Check if pandas dataframe
+        if type(df) != pd.core.frame.DataFrame:
+            raise exceptions.PlotlyError("Dataframe not inputed. Please " 
+                                         "use a pandas dataframe to pro" 
+                                         "duce a scatterplot matrix.")
+
+        # Check if dataframe is 1 row or less.
+        if len(df.columns) <= 1:
+            raise exceptions.PlotlyError("Dataframe has only one column. To "
+                                         "use the scatterplot matrix, use at "
+                                         "least 2 columns.")
+
+        # Check that diagonal parameter is selected properly
+        if diagonal not in diagonal_choices:
+            raise exceptions.PlotlyError("Make sure diagonal is set to "
+                                         "either Scatter or Histogram.")
+
+
+        if useindex == True:
+            if index not in df:
+                raise exceptions.PlotlyError("Make sure you set the index " 
+                                             "input variable to one of the "
+                                             "column names of your matrix.")
+
+            else:
+                index_vals = df[index].values.tolist()
+
+            # Make list of column names besides index
+            for name in df:
+                if name != index:
+                    col_names.append(name)
+
+            # Populate matrix = []
+            for name in col_names:
+                matrix.append(df[name].values.tolist())
+
+            # Check if Matrix Values are either all strings or numbers
+            for vector in matrix:   
+                if ((type(vector[0]) == float) or (type(vector[0]) == int)):
+                    for entry in vector:
+                        if not ((type(entry) == float) or (type(entry) == int)):
+                            raise exceptions.PlotlyError("Error in data"
+                                                         "frame. Make sure "
+                                                         "that all entries "
+                                                         "of each column are "
+                                                         "either numbers or "
+                                                         "strings.")
```

Comment:
Just a style thing, but this would be easier to read if written:

```
raise exceptions.PlotlyError(
    'Error in dataframe. Make sure that all entries '
    'of each column are either numbers or strings.'
)
```

That goes for a bunch of these messaged.